sdk/core/azure-core/test/ut/context.cpp

```diff
@@ -93,6 +119,18 @@ TEST(Context, IsCancelled)
   EXPECT_TRUE(c2.IsCancelled());
 }
 
+TEST(Context, ThrowIfCancelled)
+{
+  auto duration = std::chrono::milliseconds(500);
+  auto deadline = std::chrono::system_clock::now() + duration;
+
+  Context context;
+  auto c2 = context.WithDeadline(deadline);
```

Comment:
Also test cancelling a context that's created `WithValue()` instead of with deadline.

Reply:
And we need coverage for other public APIs, like `CancelWhen()` before closing the issue.

Reply:
I've moved CancelWhen to be private.  It is not needed by any of our current scenarios.  We can look to make it public in the future if customer demand materializes.